### PR TITLE
CSS variables refactor

### LINKED ISSUE #614: All modules: remove duplicate selectors and normalize variables across ds4/ds6
Recently we refactored the actionable and badge modules to better leverage LESS variables. It made the resulting code A LOT easier to reason with.  

Here are the details:

https://github.com/eBay/skin/issues/502

Let's sweep through the remaining modules, starting with the easiest first, and get them all following the same pattern. I think it's a great time to start on this now, and we don't have to do them all in the same release.

### PR DESCRIPTION
**Pull request by seangates**
## Description

**Updated the following components** with the pattern outlined in #502 
- Switch
- <strike>Breadcrumb</strike>
- Checkbox
- Color

## Context
This will help with maintenance, and especially for new folks working on Skin for the first time. Additionally, it will prepare us for using CSS variables, which is right around the corner!

## References
Closes #614 

## Screenshots
n/a

### CODE REVIEW

**Comment by ianmcburnie** [comment]
Looking good!

**Comment by seangates** [comment]
In my estimation, it's trivial to add an "else" statement. It was not at all a refactor or change in rules.

**Comment by ianmcburnie** [comment]
Sure, it was just a little hard for me to pick out in the diff.

**Comment by ianmcburnie** [comment]
Probably a good time to stop and review this now. Four is enough for one PR.

**Comment by seangates** [comment]
@ianmcburnie Done. Converted to a full-fledged PR.

File: dist/icon/background/ds4/background.css [hunk 1]
```diff
@@ -1,6 +1,5 @@
 span.icon {
   background-repeat: no-repeat;
-  background-size: contain;
```
Comment (ianmcburnie) [inline dist/icon/background/ds4/background.css]: @seangates Just checking that this removal is intended...

Comment (seangates) [reply dist/icon/background/ds4/background.css]: Oooh, yeah. That's not intended. I'll take a look.

Comment (seangates) [reply dist/icon/background/ds4/background.css]: Okay, so this is correct. Note the `background-size` a few rules down.

https://github.com/eBay/skin/blob/890af66cfea753a018b9be741b0fcd359300b8f2/dist/breadcrumb/ds4/breadcrumb.css#L19

**Comment by ianmcburnie** [comment]
It's kind of hard to see what's going on because of the background size change. A lot of files are being touched. I think we should isolate that set of changes to it's own PR if we can.

File: dist/pagination/ds4/pagination.css [hunk 2]
```diff
@@ -104,38 +103,44 @@
   height: 17px;
   width: 10px;
   background-size: 10px 17px;
+  background-size: contain;
```
Comment (ianmcburnie) [inline dist/pagination/ds4/pagination.css]: Is the double background size intended? I know we already had some issues with this across our existing code anyway, but want to make sure this is intended. I do feel like these background size changes should be confined to their own PR (separate from switch, color and checbox refactor).

**Comment by seangates** [comment]
Merged instead of squashed. Gah!
